tools/system_libs.py

```diff
@@ -618,6 +625,28 @@ def get_depends(self):
     return depends
 
 
+class libsockets(MuslInternalLibrary, MTLibrary):
```

Comment:
If you are always going to link `libsockets` or `libsockets_proxy`, have you considered making them different versions of the same library?

Reply:
I looked at that machinery, but it was not obvious to me how that works.

Reply:
@quantum5 maybe some docs would help explain how to use features like different library versions?

Reply:
Added some explanation in #9215